modules/interactiveOffersBidAdapter.js

```diff
@@ -0,0 +1,149 @@
+        utils.logWarn('tmax must be a valid numeric ID');
+        ret = false;
+      }
+    } else {
+      utils.logWarn('invalid request');
+      ret = false;
+    }
+    return ret;
+  },
+  buildRequests: function(validBidRequests, bidderRequest) {
+    let payload = parseRequestPrebidjsToOpenRTB(bidderRequest);
+    return {
+      method: 'POST',
+      url: ENDPOINT,
+      data: JSON.stringify(payload),
+      bidderRequest: bidderRequest
+    };
+  },
+
+  interpretResponse: function(response, request) {
+    let bidResponses = [];
+    if (response.body && response.body.length) {
+      bidResponses = parseResponseOpenRTBToPrebidjs(response.body);
+    }
+    return bidResponses;
+  }
+};
+
+function parseRequestPrebidjsToOpenRTB(prebidRequest) {
+  let pageURL = window.location.href;
+  let domain = window.location.hostname;
+  let secure = (window.location.protocol == 'https:' ? 1 : 0);
+  let openRTBRequest = JSON.parse(JSON.stringify(DEFAULT['OpenRTBBidRequest']));
+  openRTBRequest.id = prebidRequest.auctionId;
+
+  openRTBRequest.site = JSON.parse(JSON.stringify(DEFAULT['OpenRTBBidRequestSite']));
+  openRTBRequest.site.id = domain;
+  openRTBRequest.site.name = domain;
+  openRTBRequest.site.domain = domain;
+  openRTBRequest.site.page = pageURL;
+  openRTBRequest.site.ref = prebidRequest.refererInfo.referer;
+
+  openRTBRequest.site.publisher = JSON.parse(JSON.stringify(DEFAULT['OpenRTBBidRequestSitePublisher']));
+  openRTBRequest.site.publisher.id = 0;
+  openRTBRequest.site.publisher.name = config.getConfig('publisherDomain');
+  openRTBRequest.site.publisher.domain = domain;
+  openRTBRequest.site.publisher.domain = domain;
+
+  openRTBRequest.site.content = JSON.parse(JSON.stringify(DEFAULT['OpenRTBBidRequestSiteContent']));
+
+  openRTBRequest.source = JSON.parse(JSON.stringify(DEFAULT['OpenRTBBidRequestSource']));
+  openRTBRequest.source.fd = 0;
+  openRTBRequest.source.tid = prebidRequest.auctionId;
+  openRTBRequest.source.pchain = '';
+
+  openRTBRequest.device = JSON.parse(JSON.stringify(DEFAULT['OpenRTBBidRequestDevice']));
+
+  openRTBRequest.user = JSON.parse(JSON.stringify(DEFAULT['OpenRTBBidRequestUser']));
+
+  openRTBRequest.imp = [];
+  prebidRequest.bids.forEach(function(bid, impId) {
+    impId++;
+    let imp = JSON.parse(JSON.stringify(DEFAULT['OpenRTBBidRequestImp']));
+    imp.id = impId;
+    imp.secure = secure;
+    imp.tagid = bid.bidId;
+
+    openRTBRequest.site.publisher.id = openRTBRequest.site.publisher.id || bid.params.pubid;
+    openRTBRequest.tmax = openRTBRequest.tmax || bid.params.tmax || 0;
+
+    Object.keys(bid.mediaTypes).forEach(function(mediaType) {
+      if (mediaType == 'banner') {
+        imp.banner = JSON.parse(JSON.stringify(DEFAULT['OpenRTBBidRequestImpBanner']));
+        imp.banner.w = 0;
+        imp.banner.h = 0;
+        imp.banner.format = [];
+        bid.mediaTypes[mediaType].sizes.forEach(function(adSize) {
+          if (!imp.banner.w) {
```

Comment:
only check for `w` here?

Reply:
As both w and h are fully controlled by the script, and both are set at the same time, I can only look at one of them and know what the other is, in this case the script is grabbing the first size available.
It will be changed in the next commit